src/tests/stencil_tests.cpp

```diff
@@ -0,0 +1,169 @@
+#include "stencil_tests.h"
+
+#include <pbkit/pbkit.h>
+
+#include "../shaders/precalculated_vertex_shader.h"
+#include "../test_host.h"
+#include "debug_output.h"
+#include "nxdk_ext.h"
+#include "pbkit_ext.h"
+#include "vertex_buffer.h"
+
+constexpr StencilTests::StencilParams kStencilParams[] = {
+    // Stencil enable, depth disable
+    {NV097_SET_STENCIL_OP_V_ZERO, "ZERO", true, false, 0xFF, 0},
+    {NV097_SET_STENCIL_OP_V_REPLACE, "REPLACE", true, false, 0x00, 1},
+    // Stencil disable, depth enable
+    {NV097_SET_STENCIL_OP_V_ZERO, "ZERO", false, true, 0xFF, 0},
+    {NV097_SET_STENCIL_OP_V_REPLACE, "REPLACE", false, true, 0x00, 1},
+    // Stencil enable, depth enable
+    {NV097_SET_STENCIL_OP_V_ZERO, "ZERO", true, true, 0xFF, 0},
+    {NV097_SET_STENCIL_OP_V_REPLACE, "REPLACE", true, true, 0x00, 1},
+    // Stencil disable, depth disable
+    {NV097_SET_STENCIL_OP_V_ZERO, "ZERO", false, false, 0xFF, 0},
+    {NV097_SET_STENCIL_OP_V_REPLACE, "REPLACE", false, false, 0x00, 1}
+};
+
+StencilTests::StencilTests(TestHost &host, std::string output_dir)
+    : TestSuite(host, std::move(output_dir), "Stencil") {
+  for (auto param : kStencilParams) {
+      AddTestEntry(param);
+  }
+}
+
+void StencilTests::Initialize() {
+  TestSuite::Initialize();
+  SetDefaultTextureFormat();
+
+  auto shader = std::make_shared<PrecalculatedVertexShader>();
+  host_.SetVertexShaderProgram(shader);
+}
+
+void StencilTests::Test(const StencilParams &params) {
+  int quadSize = 200;
+
+  host_.SetupControl0(true);
+  host_.SetSurfaceFormat(host_.GetColorBufferFormat(),
+                         static_cast<TestHost::SurfaceZetaFormat>(NV097_SET_SURFACE_FORMAT_ZETA_Z24S8),
+                         host_.GetFramebufferWidth(), host_.GetFramebufferHeight());
+  host_.SetDepthBufferFloatMode(false);
```

Comment:
Nit: this should be guaranteed in the superclass initializer already and can be omitted.

Reply:
Removing the `SetSurfaceFormat` call seems to break my test. I think the setting in the initializer is being replaced by `TestSuite::Initialize` with z16: https://github.com/abaire/nxdk_pgraph_tests/blob/main/src/tests/test_suite.cpp#L69-L70

Reply:
Yeah, I just mean the float mode line

Reply:
That would make more sense, sorry about that. Fixed.